scripts/run-ci-e2e-tests.js

```diff
@@ -0,0 +1,170 @@
+  echo('Failed to execute react-native init');
+  echo('Most common reason is npm registry connectivity, try again');
+  exit(cleanup(1));
+}
+cd('EndToEndTest');
+
+if (args.indexOf('--android') !== -1) {
+  echo('Running an Android e2e test');
+  echo('Installing e2e framework');
+  if(exec('npm install --save-dev appium@1.5.1 mocha@2.4.5 wd@0.3.11 colors@1.0.3 pretty-data2@0.40.1', {silent: true}).code) {
+    echo('Failed to install appium');
+    exit(cleanup(1));
+  }
+  cp(`${SCRIPTS}/android-e2e-test.js`, 'android-e2e-test.js');
+  cd('android');
+  echo('Downloading Maven deps');
+  exec('./gradlew :app:copyDownloadableDepsToLibs');
+  // Make sure we installed local version of react-native
+  if (!test('-e', path.basename(MARKER_ANDROID))) {
+    echo('Android marker was not found, react native init command failed?');
+    exit(cleanup(1));
+  }
+  cd('..');
+  exec('keytool -genkey -v -keystore android/keystores/debug.keystore -storepass android -alias androiddebugkey -keypass android -dname "CN=Android Debug,O=Android,C=US"');
+
+  echo(`Starting packager server, ${SERVER_PID}`);
+  const appiumProcess = spawn('node', ['./node_modules/.bin/appium']);
+  APPIUM_PID = appiumProcess.pid;
+  echo(`Starting appium server, ${APPIUM_PID}`);
+  echo('Building app');
+  if (exec('buck build android/app').code) {
+    echo('could not execute Buck build, is it installed and in PATH?');
+    exit(cleanup(1));
+  }
+  let packagerEnv = Object.create(process.env);
+  packagerEnv.REACT_NATIVE_MAX_WORKERS = 1;
+  // shelljs exec('', {async: true}) does not emit stdout events, so we rely on good old spawn
+  const packagerProcess = spawn('npm', ['start'], {
+    // stdio: 'inherit',
+    env: packagerEnv
+  });
+  SERVER_PID = packagerProcess.pid;
+  // wait a bit to allow packager to startup
+  exec('sleep 5s');
+  echo('Executing android e2e test');
+  if(exec('node node_modules/.bin/_mocha android-e2e-test.js').code) {
+    exit(cleanup(1));
+  }
+}
+
+if (args.indexOf('--ios') !== -1) {
+  echo('Running an iOS app');
+  cd('ios');
+  // Make sure we installed local version of react-native
+  if (!test('-e', path.join('EndToEndTest', path.basename(MARKER_IOS)))) {
+    echo('iOS marker was not found, react native init command failed?');
+    exit(cleanup(1));
+  }
+  // shelljs exec('', {async: true}) does not emit stdout events, so we rely on good old spawn
+  let packagerEnv = Object.create(process.env);
+  packagerEnv.REACT_NATIVE_MAX_WORKERS = 1;
+  const packagerProcess = spawn('npm', ['start'], 
+  {
+    stdio: 'inherit',
+    env: packagerEnv
+  });
+  SERVER_PID = packagerProcess.pid;
+  echo(`Starting packager server, ${SERVER_PID}`);
+  exec('sleep 5s');
+  // prepare cache to reduce chances of possible red screen "Can't fibd variable __fbBatchedBridge..."
+  exec('response=$(curl --write-out %{http_code} --silent --output /dev/null localhost:8081/index.ios.bundle?platform=ios)');
```

Comment:
Android e2e test is smarter.
If it sees a red screen it tries to reload the app and then it works.
Both solutions are a hack that indicate a race condition we have in the framework.